File: adventofcode/day3/bin/day3.swift
func ==(l: day_two, r: day_two) -> Bool {
  return l === r
}
class day_two : Equatable  { 
  class func distance(data : Int) -> Int {
    /** unused:  let size : Int = 1   **/ 
    var side : Int = 1
    var total : Int = 1
    var last_total : Int = 1
    var n : Int = 1
    while (total < data) {
      last_total = total;
      side = side + 2;
      total = total + ((side - 1) * 4);
      n = n + 1;
    }
    let dist : Int = (data - last_total) - 1
    let sideStep : Int = side - 1
    let pos : Int = dist % sideStep
    var step_ort : Int = 0
    let halfway : Int = Int((sideStep / 2))
    if ( pos < halfway ) {
      step_ort = (halfway - 1) - pos;
    } else {
      step_ort = (pos - halfway) + 1;
    }
    print((("total steps for " + String(data)) + " == ") + String((step_ort + halfway)))
    return step_ort + halfway;
  }
}
func __main__swift() {
  _ = day_two.distance(data : 9)
  _ = day_two.distance(data : 10)
  _ = day_two.distance(data : 11)
  _ = day_two.distance(data : 12)
  _ = day_two.distance(data : 17)
  _ = day_two.distance(data : 23)
  _ = day_two.distance(data : 24)
  _ = day_two.distance(data : 1024)
  _ = day_two.distance(data : 289326)
}
// call the main function
__main__swift()

File: adventofcode/day3/bin/day3.go
package main
import (
  "strings"
  "strconv"
  "fmt"
)

type GoNullable struct { 
  value interface{}
  has_value bool
}

type day_two struct { 
}

func CreateNew_day_two() *day_two {
  me := new(day_two)
  return me;
}
func day_two_static_distance(data int64) int64 {
  /** unused:  size*/
  var side int64= int64(1);
  var total int64= int64(1);
  var last_total int64= int64(1);
  var n int64= int64(1);
  for total < data {
    last_total = total; 
    side = side + int64(2); 
    total = total + ((side - int64(1)) * int64(4)); 
    n = n + int64(1); 
  }
  var dist int64= (data - last_total) - int64(1);
  var sideStep int64= side - int64(1);
  var pos int64= dist % sideStep;
  var step_ort int64= int64(0);
  var halfway int64= int64((sideStep / int64(2)));
  if  pos < halfway {
    step_ort = (halfway - int64(1)) - pos; 
  } else {
    step_ort = (pos - halfway) + int64(1); 
  }
  fmt.Println( strings.Join([]string{ ((strings.Join([]string{ "total steps for ",strconv.FormatInt(data, 10) }, "")) + " == "),strconv.FormatInt((step_ort + halfway), 10) }, "") )
  return step_ort + halfway
}
func main() {
  day_two_static_distance(int64(9));
  day_two_static_distance(int64(10));
  day_two_static_distance(int64(11));
  day_two_static_distance(int64(12));
  day_two_static_distance(int64(17));
  day_two_static_distance(int64(23));
  day_two_static_distance(int64(24));
  day_two_static_distance(int64(1024));
  day_two_static_distance(int64(289326));
}
